src/connectors/meraki_devices.py

```diff
@@ -0,0 +1,194 @@
+    options['network_id_whitelist'] = options.get('network_id_whitelist', '').split(',')
+
+    comment = yaml_dump(module='meraki_devices', **options)
+
+    db.create_table(name=landing_table_client,
+                    cols=LANDING_TABLE_COLUMNS_CLIENT, comment=comment)
+    db.execute(f'GRANT INSERT, SELECT ON {landing_table_client} TO ROLE {SA_ROLE}')
+    
+    db.create_table(name=landing_table_device,
+                    cols=LANDING_TABLE_COLUMNS_DEVICE, comment=comment)
+    db.execute(f'GRANT INSERT, SELECT ON {landing_table_device} TO ROLE {SA_ROLE}')
+    return {
+        'newStage': 'finalized',
+        'newMessage': "Meraki ingestion tables created!",
+    }
+
+
+def ingest(table_name, options):
+    ingest_type = ''
+    if (table_name.endswith('CONNECTION_CLIENT')) or (table_name.endswith('connection_client')):
+        ingest_type = 'client'
+    else:
+        ingest_type = 'device'
+    
+    landing_table = f'data.{table_name}'
+
+    timestamp = datetime.utcnow()
+    api_key = options['api_key']
+    whitelist = options['network_id_whitelist']
+
+    for network in whitelist:
+        
+        try:
+            devices = get_data(f"https://api.meraki.com/api/v0/networks/{network}/devices", api_key)
+        except requests.exceptions.HTTPError as e:
+            log.error(f"{network} not accessible, ")
+            log.error(e)
+            continue
+
+        if (ingest_type == 'device'):
+            db.insert(
+                landing_table,
+                values=[(
+                    timestamp,
+                    device,
+                    device.get('serial',None),
+                    device.get('address',None),
+                    device.get('name',None),
+                    device.get('networkId',None),
+                    device.get('model',None),
+                    device.get('mac',None),
+                    device.get('lanIp',None),
+                    device.get('wan1Ip',None),
+                    device.get('wan2Ip',None),
+                    device.get('tags',None),
+                    device.get('lng',None),
+                    device.get('lat',None),
+                ) for device in devices],
+                select=db.derive_insert_select(LANDING_TABLE_COLUMNS_DEVICE),
+                columns=db.derive_insert_columns(LANDING_TABLE_COLUMNS_DEVICE)
+            )
+            log.info(f'Inserted {len(devices)} rows ({landing_table}).')
+            yield len(devices)
+
+        else:
+            
+            for device in devices:
+                serial_number = device['serial']
+                
+                try:
+                    clients = get_data(f"https://api.meraki.com/api/v0/devices/{serial_number}/clients", api_key)
+                except requests.exceptions.HTTPError as e:
+                    log.error(f"{network} not accessible, ")
+                    log.error(e)
+                    continue
+                
+                db.insert(
```

Comment:
sorry I wasn't more clear on this. could you please replace this db.insert expression with --


                vlan = client.get('vlan') or None
                usage_sent = client.get('usage', {}).get('sent') or None
                usage_recv = client.get('usage', {}).get('recv') or None
                db.insert(
                    landing_table,
                    values=[(
                        timestamp,
                        client,
                        client.get('id', None),
                        client.get('mac', None),
                        client.get('description', None),
                        client.get('mdnsName', None),
                        client.get('dhcpHostname', None),
                        client.get('ip', None),
                        client.get('switchport', None),
                        vlan,
                        usage_sent,
                        usage_recv,
                        serial_number,
                    ) for client in clients],
                    select=db.derive_insert_select(LANDING_TABLE_COLUMNS_CLIENT),
                    columns=db.derive_insert_columns(LANDING_TABLE_COLUMNS_CLIENT)
                )

I think it will do the same thing but more succinctly.

to clarify `x or y` will return `y` if `x`  if "falsy", i.e. if `bool(x) is False`, and `dict`'s `.get` returns None as the default value if the key is not present

Reply:
also, are you sure `vlan` needs to be treated differently than `ip` or `switchport`? this seems off.

Reply:
Got it, interesting. If you're sure that the others never return `''` then lgtm. Adding a comment like --

                    client.get('vlan') or None,  # vlan sometimes set to ''

might help to future folks, as well

Reply:
I've never seen that occur with usage:recv or usage:sent, but I think it'll be best if we used the same approach for those instances too. Thank you for this feedback!

Reply:
sgtm